### DISCUSSION TITLE
Option to use Typst as math-mode markdown language

### DISCUSSION THREAD

**Discussion by sverrirarnors** [jgm/pandoc#10381] [Ideas]
As I understand it, Pandoc treats text within `$$` as TeX math.

As Typst is gaining traction (fx. now bundled with Quarto), it would be nice to have an option to make Pandoc handle text within `$$` as Typst math instead.

Currently, this can be _kinda_ solved in Quarto with raw blocks (see https://github.com/quarto-dev/quarto-cli/discussions/11368), but a native solution would be great.

**Comment by jgm** [comment]
I believe you could achieve this with a Lua filter.  The filter would just match Math elements and use `pandoc.read` and `pandoc.write` to read the math as typst and write it as latex (extracting the inline content from the pandoc document generated by `write`).  Of course, this would be limited to typst math that pandoc/texmath can parse.

If you want to remove that limitation, but confine your output to typst, then your filter could simply convert Math elements into RawInline "typst" elements, which would pass through to typst verbatim.

**Reply by sverrirarnors** [reply]
Thanks for the suggestion! This Lua filter (specifically for quarto) was suggested here: https://github.com/quarto-dev/quarto-cli/discussions/11368#discussioncomment-11228593

```lua
-- Inline Math
Math = function(m)
  if not quarto.doc.is_format('typst') then
    return nil
  end
  if m.mathtype == "InlineMath" then
    return pandoc.RawInline('typst', '$'.. m.text .. '$')
  end
end

-- DisplayMath
-- We want to replace all Para by a RawBlock
Para = function(p)
  if not quarto.doc.is_format('typst') then
    return nil
  end
  if p.content[1].t == "Math" and p.content[1].mathtype == "DisplayMath" then
    return pandoc.RawBlock('typst', '$' .. p.content[1].text .. '$')
  end
end

```

This passes the math content through verbatim as you suggested. It works well, but would it make sense to consider having this kind of Typst math handling as a built-in option in Pandoc? That way, a user wouldn't have to include the filter in every document.

**Reply by jgm** [reply]
Math is stored in the AST as latex.  So if we had this as an option, it would require converting typst math to latex, and this can't always be done reliably.  For this reason I'm not too enthusiastic about the idea.

**Comment by tarleb** [comment]
See also: https://fosstodon.org/@pandoc/112926076691256397
